packages/remote-core/src/serialization/FlowThreadSerialization.ts

```diff
@@ -27,6 +27,9 @@ export class FlowThreadSerialization extends ThreadSerializationStructuredClone
           if (isObjectType(val)) {
             return serialize({ ...val });
           }
+          if (isArray(val)) {
```

Comment:
This is not required, because it is already handled by the base serializer (see `isSerializableByBase`).